libs/traits/src/lib.rs

```diff
@@ -321,6 +368,23 @@ pub trait TrancheCurrency<PoolId, TrancheId> {
 	fn of_tranche(&self) -> TrancheId;
 }
 
+impl TrancheCurrency<PoolId, TrancheId> for TrancheCurrencyT {
+	fn generate(pool_id: PoolId, tranche_id: TrancheId) -> Self {
+		Self {
+			pool_id,
+			tranche_id,
+		}
+	}
+
+	fn of_pool(&self) -> PoolId {
+		self.pool_id
+	}
+
+	fn of_tranche(&self) -> TrancheId {
+		self.tranche_id
+	}
+}
+
```

Comment:
This must not be here.

Reply:
The problem started when we moved `pub trait PoolMutate<...` into `cfg-traits`. This trait depends on `TrancheInput` and `PoolChanges`. Therefore, we need to depend on `cfg-types` inside `cfg-traits`.